src/extension.ts

```diff
@@ -11,20 +11,29 @@ const status: any = {};
 export function activate(context: vscode.ExtensionContext) {
     // The reporter will be initialized by the later telemetry handler.
     let reporter: TelemetryReporter = null;
-
+    let configurationDirty: boolean = true;
```

Comment:
Naming:  configurationDirty  ==> isConfigurationDirty

Also, the dirty is not appropriate here. Maybe just `update` is oaky